doc/reference/misc/timeutil.rst

```diff
@@ -0,0 +1,289 @@
+difference between TAI and UTC from 36 seconds to 37 seconds. There was
+no leap second introduced at the end of 1999, when the difference
+between TAI and UTC was only 32 seconds. The following table shows
+relevant civil and epoch times in several scales:
+
+==================== ========== =================== ======= ==============
+UTC Date             UNIX time  TAI Date            TAI-UTC UNIX Leap Time
+==================== ========== =================== ======= ==============
+1970-01-01T00:00:00Z 0          1970-01-01T00:00:08 +8      0
+1999-12-31T23:59:28Z 946684768  2000-01-01T00:00:00 +32     946684792
+1999-12-31T23:59:59Z 946684799  2000-01-01T00:00:31 +32     946684823
+2000-01-01T00:00:00Z 946684800  2000-01-01T00:00:32 +32     946684824
+2016-12-31T23:59:59Z 1483228799 2017-01-01T00:00:35 +36     1483228827
+2016-12-31T23:59:60Z undefined  2017-01-01T00:00:36 +36     1483228828
+2017-01-01T00:00:00Z 1483228800 2017-01-01T00:00:37 +37     1483228829
+==================== ========== =================== ======= ==============
+
+Functional Requirements
+-----------------------
+
+The Zephyr tick counter has no concept of leap seconds or standard time
+offsets and is a continuous time scale. However it can be relatively
+inaccurate, with drifts as much as three minutes per hour (assuming an
+RC timer with 5% tolerance).
+
+There are two stages required to support conversion between Zephyr time
+and common human time scales:
+
+* Translation between the continuous but inaccurate Zephyr time scale
+  and an accurate external stable time scale;
+* Translation between the stable time scale and the (possibly
+  discontinuous) civil time scale.
+
+The API around :c:func:`timeutil_sync_state_update()` supports the first
+step of converting between continuous time scales.
+
+The second step requires external information including schedules of
+leap seconds and local time offset changes. This may be best provided by
+an external library, and is not currently part of the time utility APIs.
+
+Selecting an External Source and Time Scale
+-------------------------------------------
+
+If an application requires civil time accuracy within several seconds
+then UTC could be used as the stable time source. However, if the
+external source adjusts to a leap second there will be a discontinuity:
+the elapsed time between two observations taken at 1 Hz is not equal to
+the numeric difference between their timestamps.
+
+For precise activities a continuous scale that is independent of local
+and solar adjustments simplifies things considerably. Suitable
+continuous scales include:
+
+-  GPS time: epoch of 1980-01-06T00:00:00Z, continuous following TAI
+   with an offset of TAI-GPS=19 s.
+-  Bluetooth mesh time: epoch of 2000-01-01T00:00:00Z, continuous
+   following TAI with an offset of -32.
+-  UNIX Leap Time: epoch of 1970-01-01T00:00:00Z, continuous following
+   TAI with an offset of -8.
+
```

Comment:
I must have missed something in the text leading up to this, but I seek a clarification of those TAI offsets?

Reply:
They're the TAI-UTC offset in effect at the corresponding epoch.  For the first two they can be determined using the leap  second table; the UNIX epoch one is calculated as shown above.